src/components/UserSelector.tsx

```diff
@@ -1,35 +1,84 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+
+type Props = {
+  onSelect: (user: User) => void,
+  selectedUser: User | null,
+};
+
+export const UserSelector: React.FC<Props> = ({
+  onSelect,
+  selectedUser,
+}) => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [showList, setShowList] = useState(false);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+
+    return () => setUsers(null);
```

Comment:
It's not necessary to clean up the state on unmounting, as the state will be automatically cleared when the component unmounts. You can remove the `return () => setUsers(null);` line.